src/components/App.tsx

```diff
@@ -208,57 +204,123 @@ console.log("unmount")
 
   const onMessageReceived = (msg: any): void => {
     // console.log('message: ', msg.content.msg_type);
-    if (msg.content.msg_type === "IMAGE"){
+    if (msg.content.msg_type === "IMAGE") {
       setState({
         ...state,
         messages: state.messages.concat({
           username: currentUser.name,
           text: msg.content.title,
+          time:
+            (new Date().getHours() % 12 || 12) +
+            ":" +
+            (new Date().getMinutes() < 10 ? "0" : "") +
+            new Date().getMinutes() +
+            (new Date().getHours() > 12 ? " PM" : " AM"),
+          date:
+            (new Date().getDate() < 10 ? "0" : "") +
+            new Date().getDate() +
+            "/" +
+            (new Date().getMonth() + 1 < 10 ? "0" : "") +
+            (new Date().getMonth() + 1) +
+            "/" +
+            new Date().getFullYear(),
           image: msg.content.media_url,
           choices: msg.content.choices,
           caption: msg.content.caption,
         }),
       });
-    }
-    else if (msg.content.msg_type === "AUDIO"){
+    } else if (msg.content.msg_type === "AUDIO") {
       setState({
         ...state,
         messages: state.messages.concat({
           username: currentUser.name,
           text: msg.content.title,
+          time:
+            (new Date().getHours() % 12 || 12) +
+            ":" +
```

Comment:
Having the code redundancy. 
Better to have a time and date getter functions which have this and use can simple call the function here

Reply:
The whole code base is written  to deal with different  types contents